py/dml/traits.py

```diff
@@ -358,14 +362,22 @@ def mktrait(site, tname, ancestors, methods, params, sessions, hooks,
                     report(EDMETH(msite, orig_trait.method_impls[name].site,
                                   name))
                     bad_methods.add(name)
+                elif explicit_decl:
+                    report(EOVERRIDEMETH(msite, orig_site, name,
+                                         'default '*overridable))
+                    bad_methods.add(name)
                 elif name not in ancestor_vtables:
                     raise ICE(msite,
                               'ancestor is overridable but not in vtable')
-
                 # Type-checking of overrides is done later, after typedefs
                 # have been populated with all template types.
                 # See Trait.typecheck_methods()
 
+        if (body is not None and not some_coll and not explicit_decl
+            and msite.provisional_enabled(provisional.explicit_method_decls)):
+            report(ENOVERRIDEMETH(msite, name, 'default '*overridable))
```

Comment:
space around `*`, here and in other places (I prefer `if` ternary but don't insist)